aspnetcore/fundamentals/minimal-apis/7.0-samples/todo-group/TodoEndpointsV2.cs

```diff
@@ -0,0 +1,106 @@
+using todo_group.Data;
+using todo_group.Services;
+
+namespace todo_group;
+
+public static class TodoEndpointsV2
```

Comment:
What is the goal in add a V2?

Reply:
V1 and V2 follow different patterns for accessing the underlying DB context. In V1 the context is directly injected into the endpoint whereas in V2 an additional abstraction (TodoService) is used to wrap the DB context. I think this is very common to use this abstraction (e.g. Repository pattern) and adopted in many applications to make the endpoint/controllers more testable. Rather than faking the context with an In-Memory implementation, we use a mock of a service. To show this, I've used `Moq` for mocking the `TodoService` in `TodoMoqTests`. Similarly for V1, I've used the In-Memory database approach found in `TodoInMemoryTests`